/**:
  ros__parameters:
    common:
      # output
      output_delta_arc_length: 0.5     #  delta arc length for output trajectory [m]
      output_backward_traj_length: 5.0  # backward length for backward trajectory from base_link [m]
    # elastic band
    elastic_band:
      option:
        enable_warm_start: true
        enable_optimization_validation: false

      common:
        num_points: 100        # number of points for optimization [-]
        delta_arc_length: 1.0  # delta arc length for optimization [m]

      clearance:
        num_joint_points: 3  # number of joint points (joint means connecting fixing and smoothing)

        clearance_for_fix: 0.0     # maximum optimizing range when applying fixing
        clearance_for_joint: 0.1   # maximum optimizing range when applying jointing
        clearance_for_smooth: 0.1  # maximum optimizing range when applying smoothing

      weight:
        smooth_weight: 1.0
        lat_error_weight: 0.001

      qp:
        max_iteration: 10000  # max iteration when solving QP
        eps_abs: 1.0e-7       # eps abs when solving OSQP
        eps_rel: 1.0e-7       # eps rel when solving OSQP

      validation: # if enable_optimization_validation is true
        max_error: 3.0 # [m]
    # nearest search
    ego_nearest_dist_threshold: 3.0 # [m]
    ego_nearest_yaw_threshold: 1.046 # [rad] = 60 [deg]
    # replanning & trimming trajectory param outside algorithm
    replan:
      enable: true  # if true, only perform smoothing when the input changes significantly
      max_path_shape_around_ego_lat_dist: 2.0  # threshold of path shape change around ego [m]
      max_path_shape_forward_lon_dist: 100.0   # forward point to check lateral distance difference [m]
      max_path_shape_forward_lat_dist: 0.1     # threshold of path shape change around forward point [m]
      max_ego_moving_dist: 5.0                 # threshold of ego's moving distance for replan [m]
      # make max_goal_moving_dist long to keep start point fixed for pull over
      max_goal_moving_dist: 15.0               # threshold of goal's moving distance for replan [m]
      max_delta_time_sec: 1.0                  # threshold of delta time for replan [second]
